core/src/services/cloudflare_kv/backend.rs

```diff
@@ -0,0 +1,444 @@
+                    Error::new(
+                        crate::ErrorKind::Unexpected,
+                        &format!("failed to parse error response: {}", e),
+                    )
+                })?;
+                if !response.success {
+                    return Err(Error::new(
+                        crate::ErrorKind::Unexpected,
+                        &String::from_utf8_lossy(&body),
+                    ));
+                }
+                Ok(())
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn delete(&self, path: &str) -> Result<()> {
+        let url = format!(
+            r"https://api.cloudflare.com/client/v4/accounts/{}/storage/kv/namespaces/{}",
+            self.account_id, self.namespace_id
+        );
+        let url = format!("{}/values/{}", url, path);
+        let mut req = Request::delete(&url);
+        req = req.header(header::CONTENT_TYPE, "application/json");
+        let mut req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        req = self.load_token(req)?;
+        let resp = self.client.send(req).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK => {
+                let body = resp.into_body().bytes().await?;
+                let response: CfKvResponse = serde_json::from_slice(&body).map_err(|e| {
+                    Error::new(
+                        crate::ErrorKind::Unexpected,
+                        &format!("failed to parse error response: {}", e),
+                    )
+                })?;
+                if !response.success {
+                    return Err(Error::new(
+                        crate::ErrorKind::Unexpected,
+                        &String::from_utf8_lossy(&body),
+                    ));
+                }
+                Ok(())
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn scan(&self, path: &str) -> Result<Vec<String>> {
+        let url = format!(
+            r"https://api.cloudflare.com/client/v4/accounts/{}/storage/kv/namespaces/{}",
+            self.account_id, self.namespace_id
+        );
+        let mut url = format!("{}/keys", url);
+        if path.len() > 0 {
+            url = format!("{}?prefix={}", url, path);
+        }
+        let mut req = Request::get(&url);
+        req = req.header(header::CONTENT_TYPE, "application/json");
+        let mut req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        req = self.load_token(req)?;
+        let resp = self.client.send(req).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK => {
+                let body = resp.into_body().bytes().await?;
+                let response: CfKvScanResponse = serde_json::from_slice(&body).map_err(|e| {
```

Comment:
We should continuously send cursor requests until we retrieve the entire content. Improvements can be made in the next PR or through a new kv API.

Reply:
I am not family with this api. According to this [condition](https://github.com/cloudflare/miniflare/blob/f919a2eaccf30d63f435154969e4233aa3b9531c/packages/kv/src/namespace.ts#L413),  we can check if cursor is empty to know when to end it. But I am confusing about the result_info in scanresponse, do you think it should be set as option？

Reply:
> maybe we can check if cursor is empty to know when to end it？

Yes.



> And I am confused about the result_info in scanresponse, do you think it should be set as option？

This PR is large enough that we can discuss the detail in next PR.